tests/test_io_xarray.py

```diff
@@ -525,3 +525,69 @@ def test_xarray_reader_no_dims():
         assert img.height == 256
         assert img.band_names == ["value"]
         assert img.dataset_statistics == ((arr.min(), arr.max()),)
+
+        img = dst.part((-160, -80, 160, 80))
+        assert img.count == 1
+        assert img.width == 33
+        assert img.height == 33
+        assert img.band_names == ["value"]
+        assert img.dataset_statistics == ((arr.min(), arr.max()),)
+
+
+def test_xarray_reader_Y_axis():
+    """test XarrayReader with 2D dataset."""
+    arr = numpy.arange(0, 33 * 35).reshape(1, 33, 35)
```

Comment:
A descriptive comment might be helpful here
```suggestion
    # Create a DataArray where the y coordinates are in increasing order
    # (this is the opposite of typical raster data)
    # This array will have a positive y resolution in the affine transform
    # and the data values increase with the y coordinates
    arr = numpy.arange(0, 33 * 35).reshape(1, 33, 35)
```